documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,505 @@
+Based on configured goals and capacities, it generates an optimization proposal for balancing partition distribution across brokers, which is reflected in the status of the `KafkaRebalance` resource.
+. The optimization proposal is approved or rejected (manually or automatically) based on its alignment with cluster management goals.
+. If approved, the Executor applies the optimization proposal to rebalance the Kafka cluster. 
+This involves reassigning partitions and redistributing workload across brokers according to the approved proposal.
+
+.Cruise Control optimization process
+image:kafka-concepts-cruise-control.png[Cruise Control process]
+
+Optimization proposals comprise separate partition reassignment commands.
+When you approve a proposal, the Cruise Control server applies these commands to the Kafka cluster.
+
+A partition reassignment command consists of either of the following types of operations:
+
+* Partition movement: Involves transferring the partition replica and its data to a new location. Partition movements can take one of two forms:
+** Inter-broker movement: The partition replica is moved to a log directory on a different broker.
+** Intra-broker movement: The partition replica is moved to a different log directory on the same broker.
+
+* Leadership movement: Involves switching the leader of the partition's replicas.
+
+Cruise Control issues partition reassignment commands to the Kafka cluster in batches.
+The performance of the cluster during the rebalance is affected by the number of each type of movement contained in each batch.
+
+[id='con-optimization-proposals-modes-{context}']
+=== Rebalancing modes
+
+Proposals for rebalances can be generated in three modes, which are specified using the `spec.mode` property of the `KafkaRebalance` custom resource.
+
+`full` mode:: The `full` mode runs a full rebalance by moving replicas across all the brokers in the cluster.
+This is the default mode if the `spec.mode` property is not defined in the `KafkaRebalance` custom resource.
+
+`add-brokers` mode:: The `add-brokers` mode is used after scaling up a Kafka cluster by adding one or more brokers.
+Normally, after scaling up a Kafka cluster, new brokers are used to host only the partitions of newly created topics.
+If no new topics are created, the newly added brokers are not used and the existing brokers remain under the same load.
+By using the `add-brokers` mode immediately after adding brokers to the cluster, the rebalancing operation moves replicas from existing brokers to the newly added brokers.
+You specify the new brokers as a list using the `spec.brokers` property of the `KafkaRebalance` custom resource.
+
+`remove-brokers` mode:: The `remove-brokers` mode is used before scaling down a Kafka cluster by removing one or more brokers.
+If you scale down a Kafka cluster, brokers are shut down even if they host replicas.
```

Comment:
this is actually not the case by default. We have now I check which prevents you to scale down if the brokers are hosting partitions, so that you need to run rebalancing first. There is a `strimzi.io/skip-broker-scaledown-check` annotation which can be set to true to skip this check (but it's disabled by default as just explained).

Reply:
Right. I've updated this and added a note that references the section on skipping checks on scale-down operations